builtin/k8s/platform.go

```diff
@@ -1245,7 +1278,8 @@ type Config struct {
 
 	// A full resource of options to define ports for your service running on the container
 	// Defaults to port 3000.
-	Ports []map[string]string `hcl:"ports,optional"`
+	// Todo(XX): add in HCL parse logic to warn if defining ports the old way, & update docs
+	Ports []*Port `hcl:"port,block"`
```

Comment:
Agreed, we should get this sorted out in this PR if we can. I know some people are using this already so removing it will break their waypoint.hcl. Maybe we can get an auto-upgrade a `ports` stanza to `port` structs and display a depreciation warning?